### PR TITLE
Fix external links not working on Android

### LINKED ISSUE #22777: osu! does not update on Android.
### Type

Game behaviour

### Bug description

When being notified about a new update, tapping the notification just makes it disappear and does not actually download the update.

### Screenshots or videos

https://user-images.githubusercontent.com/100294874/222691469-a7f854f7-3923-469d-a64f-8c8a564401d3.mp4

Sorry about the poor quality, Android didn't crop the video right when switching from portrait to landscape.

### Version

2023.207.0

### Logs

(unable to find)

### PR DESCRIPTION
**Pull request by Joehuu**
- Fixes https://github.com/ppy/osu/issues/22777

Will also make a PR framework-side but is not necessary for fixing osu!-side.

### CODE REVIEW

File: osu.Android/AndroidManifest.xml [hunk 1]
```diff
@@ -2,4 +2,11 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android" package="sh.ppy.osulazer" android:installLocation="auto">
 	<uses-sdk android:minSdkVersion="21" android:targetSdkVersion="31" />
 	<application android:allowBackup="true" android:supportsRtl="true" android:label="osu!" android:icon="@drawable/lazer" />
+	<queries>
+		<intent>
+			<action android:name="android.intent.action.VIEW" />
+			<category android:name="android.intent.category.BROWSABLE" />
+			<data android:scheme="https" />
+		</intent>
```
Comment (bdach) [inline osu.Android/AndroidManifest.xml:10]: The same intent query should be declared for the `mailto:` scheme, as [links of that scheme are also valid for use with `OpenUrlExternally()`](https://github.com/ppy/osu-framework/blob/32a7ef0ceb8eec36f9369a2d8b70cfd0c9c98366/osu.Framework/Extensions/ExtensionMethods.cs#L371-L376). Some places use these sorts of links (most notably wiki). Specifying `mailto:` allows the links to open the mail app to send an email out.

From experimentation, specifying `http` does not seem to be necessary (`https` suffices).

**Review by bdach — CHANGES_REQUESTED** [review]

**Comment by bdach** [comment]
Would like to get this into the upcoming hotfix release if possible, so expediting application of the review would be appreciated.

**Comment by Joehuu** [comment]
I'll wait for another opinion. I might not be around for the hotfix release so feel free to apply the changes.

**Comment by bdach** [comment]
The manifest merger thing seems pretty nice and looks to work, so I've applied (with minor stylistic adjustments), thanks 👍

**Review by bdach — APPROVED** [review]

**Pull request merged**